From git-commits-head-owner@vger.kernel.org Fri Dec  8 18:09:04 2006
Date: Thu, 7 Dec 2006 18:03:27 GMT
Message-Id: <200612071803.kB7I3RV4010059@hera.kernel.org>
From: Linux Kernel Mailing List <linux-kernel@vger.kernel.org>
To: git-commits-head@vger.kernel.org
Subject: r8169: more alignment for the 0x8168
Status: RO
Content-Length: 1566
Lines: 44

Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=cc9f022d97d08e4e36d38661857991fe91447d68
Commit:     cc9f022d97d08e4e36d38661857991fe91447d68
Parent:     12d86f682e8acad8555718dc7b0082590f2365d0
Author:     Francois Romieu <romieu@fr.zoreil.com>
AuthorDate: Fri Nov 17 23:15:17 2006 +0100
Committer:  Francois Romieu <romieu@electric-eye.fr.zoreil.com>
CommitDate: Mon Dec 4 01:04:36 2006 +0100

    r8169: more alignment for the 0x8168
    
    Two thirds of packets are lost because of misalignment. Users of
    Asus laptop did apparently not notice it.
    
    Reported on Gigabyte GA-945GM-S2.
    
    Fix for http://bugzilla.kernel.org/show_bug.cgi?id=7517
    
    Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
---
 drivers/net/r8169.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index 0b57050..2379d83 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -2018,7 +2018,7 @@ static int rtl8169_alloc_rx_skb(struct p
 	if (!skb)
 		goto err_out;
 
-	skb_reserve(skb, align);
+	skb_reserve(skb, (align - 1) & (u32)skb->data);
 	*sk_buff = skb;
 
 	mapping = pci_map_single(pdev, skb->data, rx_buf_sz,
@@ -2486,7 +2486,7 @@ static inline int rtl8169_try_rx_copy(st
 
 		skb = dev_alloc_skb(pkt_size + align);
 		if (skb) {
-			skb_reserve(skb, align);
+			skb_reserve(skb, (align - 1) & (u32)skb->data);
 			eth_copy_and_sum(skb, sk_buff[0]->data, pkt_size, 0);
 			*sk_buff = skb;
 			rtl8169_mark_to_asic(desc, rx_buf_sz);

